var/spack/repos/builtin/packages/vtk-h/package.py

```diff
@@ -284,25 +221,73 @@ def create_host_config(self, spec, prefix, py_site_pkgs_dir=None):
                 else:
                     cfg.write(cmake_cache_entry("MPIEXEC",
                                                 mpiexe_bin))
+            if "+blt_find_mpi" in spec:
+                cfg.write(cmake_cache_entry("ENABLE_FIND_MPI", "ON"))
+            else:
+                cfg.write(cmake_cache_entry("ENABLE_FIND_MPI", "OFF"))
         else:
             cfg.write(cmake_cache_entry("ENABLE_MPI", "OFF"))
 
         #######################
         # CUDA
         #######################
-
         cfg.write("# CUDA Support\n")
 
         if "+cuda" in spec:
             cfg.write(cmake_cache_entry("ENABLE_CUDA", "ON"))
+            cfg.write(cmake_cache_entry("VTKm_ENABLE_CUDA", "ON"))
+            cfg.write(cmake_cache_entry("CMAKE_CUDA_HOST_COMPILER", env["SPACK_CXX"]))
         else:
             cfg.write(cmake_cache_entry("ENABLE_CUDA", "OFF"))
+            cfg.write(cmake_cache_entry("VTKm_ENABLE_CUDA", "OFF"))
+
+        #######################################################################
+        # Core Dependencies
+        #######################################################################
+
+        #######################
+        # VTK-h (and deps)
+        #######################
+
+        cfg.write("# vtk-m support \n")
+
+        if "+openmp" in spec:
+            cfg.write("# enable openmp support\n")
+            cfg.write(cmake_cache_entry("ENABLE_OPENMP", "ON"))
+
+        cfg.write("# vtk-m from spack\n")
+        cfg.write(cmake_cache_entry("VTKM_DIR", spec['vtk-m'].prefix))
```

Comment:
If this refer to the VTKm_DIR CMake var for `find_package` use instead `VTKm_ROOT. AFAIK DIR is for the cmake/ subdir and ROOT can accept the package prefix as intended here

Reply:
Good eye. But in our case VTK-h has logic that translates `VTKM_DIR` to `VTKm_DIR` in its build system.